external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts

```diff
@@ -979,22 +999,41 @@ function traceRefValueFromJSON(value: JSONTraceRefValue): RuntimeValueType {
  * @returns runtime trace value.
  */
 function traceRuntimeValueFromJSON(value: JSONTraceRuntimeValueType): RuntimeValueType {
-    if (typeof value === 'boolean'
-        || typeof value === 'number'
-        || typeof value === 'string') {
-        return String(value);
-    } else if (Array.isArray(value)) {
-        return value.map(item => traceRuntimeValueFromJSON(item));
-    } else {
+    if (
+        value.type === 'U8' ||
+        value.type === 'U16' ||
+        value.type === 'U32' ||
+        value.type === 'U64' ||
+        value.type === 'U128' ||
+        value.type === 'Bool' ||
+        value.type === 'Address') {
+        return String(value.value);
```

Comment:
Nit. Perhaps we could rename the parameter from `value` to `move_value` (`serializable_move_value`) if you follow my other renaming suggestion (or something like `json_value` if not)